src/ruvsarpur.py

```diff
@@ -258,39 +258,52 @@ def parseArguments():
   parser.add_argument("--desc", help="Displays show description text when available", action="store_true")
   
   parser.add_argument("-d", "--debug", help="Prints out extra debugging information while script is running", action="store_true")
+
+  parser.add_argument("-p","--portable", help="Saves the tv schedule and the download log in the current directory instead of {0}".format(LOG_DIR), action="store_true")
   
   return parser.parse_args()
  
+# Appends the config directory to config file names
+def appendConfigDirectory(portable, file_name):
+  if portable :
+    return "./{0}".format(file_name)
+  else:
+    return "{0}/{1}".format(LOG_DIR,file_name)
+
+
 # Saves a list of program ids to a file
-def savePreviouslyRecordedShows(pids):
+def savePreviouslyRecordedShows(pids,portable):
```

Comment:
Instead of passing the argument flag into all the methods a less error prone approach is to construct the file paths before and then pass the completed paths into the functions. That way you limit the likelyhood of errors now or later in the scripts lifetime. Please revise